lib/rules/require-curly-braces.js

```diff
@@ -72,7 +72,7 @@ module.exports.prototype = {
         if (typeIndex['case'] || typeIndex['default']) {
             file.iterateNodesByType('SwitchCase', function(node) {
                 // empty case statement
-                if (!node.consequent || node.consequent.length === 0) {
+                if (node.consequent.length === 0) {
```

Comment:
In general, my problem with the weakening of the conditions is that they were added for edge cases. If we don't have a test hitting that condition, then we should add the test, not remove the condition. 

With the removal, this would throw an error on empty case statements since `.length` wouldn't exist on `undefined`.